### PR TITLE
feat(api): Added new EnrollmentService API.

### PR DESCRIPTION
**Pull request by salvatore-coppola**

### CODE REVIEW

File: kura/org.eclipse.kura.api/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -65,7 +65,7 @@ Export-Package: org.eclipse.kura;version="1.6.0",
  org.eclipse.kura.status;version="1.0.2",
  org.eclipse.kura.system;version="1.6.0",
  org.eclipse.kura.type;version="1.1.0",
- org.eclipse.kura.usb;version="1.3.0",
+ org.eclipse.kura.usb;version="2.0.0",
```
Comment (MMaiero) [inline kura/org.eclipse.kura.api/META-INF/MANIFEST.MF]: Why these changes if we are just adding a class in a package?

Comment (MMaiero) [reply kura/org.eclipse.kura.api/META-INF/MANIFEST.MF]: The new package should also be exported, I think.

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraErrorCode.java [hunk 2]
```diff
@@ -247,5 +247,11 @@ public enum KuraErrorCode {
      * 
      * @since 2.3
      */
-    UNSUPPORTED_MODEM;
+    UNSUPPORTED_MODEM,
+    /**
+     * Modem unsupported.
```
Comment (MMaiero) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraErrorCode.java]: Probably this is not correct

File: kura/org.eclipse.kura.api/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -2,10 +2,10 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: org.eclipse.kura.api
 Bundle-SymbolicName: org.eclipse.kura.api;singleton:=true
-Bundle-Version: 2.4.0.qualifier
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.api/META-INF/MANIFEST.MF:5]: There should be no need for a major version bump

File: kura/org.eclipse.kura.api/META-INF/MANIFEST.MF (hunk 1 above)
Comment (salvatore-coppola) [reply kura/org.eclipse.kura.api/META-INF/MANIFEST.MF]: because the baseline check plugin gave me an error preventing the build.

File: kura/org.eclipse.kura.api/META-INF/MANIFEST.MF (hunk 3 above)
Comment (salvatore-coppola) [reply kura/org.eclipse.kura.api/META-INF/MANIFEST.MF:5]: baseline check plugin said me it is required.

File: kura/org.eclipse.kura.api/META-INF/MANIFEST.MF (hunk 1 above)
Comment (MMaiero) [reply kura/org.eclipse.kura.api/META-INF/MANIFEST.MF]: Do you have the same error with develop?

Comment (salvatore-coppola) [reply kura/org.eclipse.kura.api/META-INF/MANIFEST.MF]: in develop i haven't the same problem, i try to revert and see if the error persist.

File: kura/org.eclipse.kura.api/META-INF/MANIFEST.MF (hunk 3 above)
Comment (salvatore-coppola) [reply kura/org.eclipse.kura.api/META-INF/MANIFEST.MF:5]: Solved, the problem appears only building in Eclipse, but after a full build it went away.

File: kura/org.eclipse.kura.api/META-INF/MANIFEST.MF (hunk 1 above)
Comment (salvatore-coppola) [reply kura/org.eclipse.kura.api/META-INF/MANIFEST.MF]: Solved, the problem appears only building in Eclipse, but after a full build it went away.

**Review by nicolatimeus — APPROVED** [review]
